web/ui/react-app/src/hooks/useFetches.tsx

```diff
@@ -0,0 +1,36 @@
+import { useState, useEffect } from 'react';
+
+const useFetches = <R extends any>(urls: string[], options?: RequestInit) => {
```

Comment:
Btw., this API turned out to be a bit dangerous: when you want to use it to just load a single endpoint and do `useFetches([myEndpoint])`, the page ends up in an infinite fetch loop and you can't even close the tab anymore. E.g. try changing the `Status` handler function to:

```typescript
const Status = () => {
  const ep0 = useFetch<StatusPageState>([ENDPOINTS[0]]);

  return (
    <h2>Hello</h2>
  )
};
```

It will load the runtimeinfo endpoint in an infinite loop, I guess because React thinks it's a new array every time? I could solve it by creating the single-element array as a const outside of the function and reusing that. Not sure how that can be improved for a multi-fetch function though.

Reply:
Yep, I found this yesterday when writing tests for it. The same case is with the second argument - options. Try ```const ep0 = useFetch<StatusPageState>(ENDPOINTS, {});```

Reply:
Reading more, I wonder if we are correctly obeying the rules of hooks? https://reactjs.org/docs/hooks-overview.html#rules-of-hooks

"Only call Hooks at the top level. Don’t call Hooks inside loops, conditions, or nested functions."

I mean, it seems to work...

Reply:
From the more verbose explanation at https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level it makes sense that we don't have a direct problem here, since we don't have variability in ordering or similar, but it's harder to ensure this way of course.

Reply:
I think we are not breaking any of the rules so far but yeah hooks are a bit dangerous